management/src/main/scala/akka/management/scaladsl/AkkaManagement.scala

```diff
@@ -144,17 +144,14 @@ final class AkkaManagement(implicit private[akka] val system: ExtendedActorSyste
 
         val combinedRoutes = prepareCombinedRoutes(effectiveProviderSettings)
 
-        val connectionContext =
-          effectiveProviderSettings.httpsConnectionContext.getOrElse(Http().defaultServerHttpContext)
+        val baseBuilder = Http().newServerAt(effectiveBindHostname, effectiveBindPort)
+          .withSettings(ServerSettings(system).withRemoteAddressHeader(true))
 
-        val serverFutureBinding =
-          Http().bindAndHandle(
-            RouteResult.route2HandlerFlow(combinedRoutes),
-            effectiveBindHostname,
-            effectiveBindPort,
-            connectionContext = connectionContext,
-            settings = ServerSettings(system).withRemoteAddressHeader(true)
-          )
+        val securedBuilder = effectiveProviderSettings.httpsConnectionContext match {
+          case Some(httpsContext) => baseBuilder.enableHttps(httpsContext)
+          case None => baseBuilder
+        }
+        val serverFutureBinding = securedBuilder.bindFlow(RouteResult.routeToFlow(combinedRoutes))
```

Comment:
Is there a reason, it's not just that:

```suggestion
        val serverFutureBinding = securedBuilder.bind(combinedRoutes)
```

Reply:
No reason. I followed the tips on the deprecation messages.